ghmirror/data_structures/redis_data_structures.py

```diff
@@ -33,7 +39,15 @@ def __init__(self):
             host=endpoint,
             port=port,
             password=password,
-            ssl=True)
+            ssl=True,
+            socket_timeout=5)
+        try:
+            self.cache.ping()
```

Comment:
Won't this add too much overhead?

Reply:
I am assuming you mean adding a ping to every connection? I wanted to test it out and see how that affects the latency. I couldn't figure out another way to test the connection. Maybe I can do something similar to what you did for offline mode and have a background thread pinging Redis. It does seem a bit unnecessary given that Elasticache is guaranteed to be [99.99%](https://aws.amazon.com/elasticache/sla/) available, but I thought maybe it would be a negligible overhead.

Reply:
Yes, benchmarking that would be great. But if we don't ping, when do we know redis is not reachable? I thought the timeout in the redis.Redis call would do it. If it doesn't, we will fail on get/set anyway, right?

Reply:
Just so we have more food, for thought, I ran this in one of the stage pods:
```
>>> from ghmirror.data_structures.redis_data_structures import RedisCache
>>> from timeit import timeit
>>> 
>>> def test1():
...     r = RedisCache()
... 
>>> def test2():
...     r = RedisCache()
...     r.cache.ping()
... 
>>> 
>>> print(timeit('test1()', globals=globals(), number=100))
0.015918640419840813
>>> 
>>> print(timeit('test2()', globals=globals(), number=100))
1.9662256371229887
```